src/btree/bt_cursor.c

```diff
@@ -762,6 +762,14 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
     } else
         ret = WT_NOTFOUND;
 
+    /*
+     * The format test program does repeatable reads testing, and wants to dump the cursor page on
+     * failure. It sets up a callback for that purpose, and we pay a cache miss per search to make
+     * that work.
+     */
+    if (session->format_private != NULL && (ret == 0 || ret == WT_NOTFOUND))
+        session->format_private(ret, session->format_private_arg);
```

Comment:
Maybe we could add a pointer to a debug structure that has a pointer to this method? It will allow us to extend it in the future more easily and it will be localized in one structure instead of scattered over the code base. Mabe something like this:
typedef struct {
    void (*format_private)(int, void *); /* Format test program private callback. */
    void *format_private_arg;
} WT_DEBUG_DATA;

struct __wt_session_impl {
    ....
    WT_DEBUG_DATA *debug_data;
    ....
};

Reply:
Got it - thanks for the context Keith.

I'm happy to keep it specific for now, and generalise if we identify a need.

@AlexeyAnisimovAU could you take a stab at moving it into the WT_CURSOR instead of the WT_SESSION, and see what it looks like?

Reply:
Yes sure, I'll give it a go.

Reply:
Here's the ticket for it: [WT-9562](https://jira.mongodb.org/browse/WT-9562)